### PR TITLE
Adding nutrient components to documentation and updating developer docs

### PR DESCRIPTION
**Pull request by mattjbr123**
For #96 and #99

### CODE REVIEW

**Comment by ThibHlln** [comment]
Thanks Matt for creating this PR. I have made comments on the developers.rst.

One overall comment I would make is that I think the Developer Guide is likely to broader than just adding new "cycles"/components to the framework. For example, I can imagine that new SpaceDomain will be developed in the future. This would belong in this section too. So what you wrote, I would probably put it in a sub-section of the Developer Guide to leave room for additional sub-sections in the future.

Otherwise, I think that all the information provided is very useful. It is great that you took the time to document what you went through when adding the Nutrient cycle to the framework.

**Comment by mattjbr123** [comment]
Thanks Thibault :)

**Comment by mattjbr123** [comment]
@ThibHlln where do I find the comments you've made?

**Comment by ThibHlln** [comment]
> @ThibHlln where do I find the comments you've made?

Sorry, I had forgotten to submit my review. :-/

File: docs/_doc_src/developers.rst [hunk 1]
```diff
@@ -9,4 +9,160 @@ capabilities of the framework.
 
 .. warning::
 
-   Not available yet.
+   Work in progress...
+
+Development Guidance
+--------------------
+The UnifHy repository is at https://github.com/unifhy-org/unifhy. The recommended approach if you wish to make your changes available to all users of UnifHy is to create a branch/fork of the repository, make the changes and create a Pull Request once done. Creating the Pull Request triggers the unit tests to run, but these can also be triggered manually by changing the status of the PR to draft and back again.
```
[the remark this answers is not in the document]
Comment (mattjbr123) [reply docs/_doc_src/developers.rst]: D'oh, I don't think I've ever got the capitilisations correct xD

File: docs/_doc_src/developers.rst [hunk 2]
```diff
@@ -9,4 +9,160 @@ capabilities of the framework.
...
+.. figure:: ../../_doc_img/framework_detailed_transfers.svg
+   :align: center
+   :alt: component transfers
+
+Each core Component of UnifHy is defined by a python class which subclasses the `Component` class. The `Component` class contains all the functionality that underpins all UnifHy Components. The only significant difference between the Components is defined by the `_inwards_info` and `_outwards_info` attributes which define the transfers each Component is set up to receive and produce. 
+
+To add new transfers to the UnifHy framework it is therefore a case of adding the name and details of the transfer to the `_outwards_info` attribute of the Component the transfer is coming from and the `_inwards_info` attribute of the Component the transfer is going to. 
+
+.. rubric:: Example
+
```
[the remark this answers is not in the document]
Comment (mattjbr123) [reply docs/_doc_src/developers.rst]: Thanks for this and all the rst syntax corrections, I was very much using the other rst files in the docs as templates, and was bound to miss some things!

File: docs/_doc_src/developers.rst [hunk 3]
```diff
@@ -9,4 +9,160 @@ capabilities of the framework.
...
+.. rubric::example
+   E.g. for a transfer from Component A running with a 15-min timestep and Component B running on an hourly timestep, the Exchanger will take either a (weighted) mean, (weighted) sum, the min or the max of the 4 timestep values of Component A's transfer to provide to Component B, depending on the value of `method`.  
+
+.. note::
+
+   Remember adding a transfer between Components in the *framework* means that any models developed to fit one or more of the affected Components of the framework now have the option to receive and make use of the transfer variable if they are sub-classing (based on) the 'to' Component, or have the option to produce the transfer variable if they are sub-classing the 'from' Component. In the example above, any models developed for the NutrientSurfaceLayer Component could make use of `transfer_x` and any models developed for the SurfaceLayer Component could produce `transfer_x`. Both models would have to be developed, or existing models adapted, to build a Model that actually made use of the new transfer.
```
[the remark this answers is not in the document]
Comment (mattjbr123) [reply docs/_doc_src/developers.rst]: You make a good point. I've adapted this note (which is now at the bottom of  'developers/transfers.rst') and added in text below it and to the start of 'developers/components.rst' to this effect. Let me know your thoughts.

[the remark this answers is not in the document]
Comment (ThibHlln) [reply docs/_doc_src/developers.rst]: Thanks Matt. Looks good to me! :-)

**Pull request merged**